src/screens/Onboarding/StepFinished.tsx

```diff
@@ -41,17 +48,58 @@ export function StepFinished() {
   const queryClient = useQueryClient()
   const agent = useAgent()
   const requestNotificationsPermission = useRequestNotificationsPermission()
+  const activeStarterPack = useActiveStarterPack()
+  const setActiveStarterPack = useSetActiveStarterPack()
+  const setHasCheckedForStarterPack = useSetHasCheckedForStarterPack()
+  const setSelectedFeed = useSetSelectedFeed()
 
   const finishOnboarding = React.useCallback(async () => {
     setSaving(true)
-
-    const {interestsStepResults, profileStepResults} = state
-    const {selectedInterests} = interestsStepResults
     try {
+      let starterPack: AppBskyGraphDefs.StarterPackView | undefined
+      let listItems: AppBskyGraphDefs.ListItemView[] | undefined
+
+      if (activeStarterPack?.uri) {
+        const spRes = await agent.app.bsky.graph.getStarterPack({
+          starterPack: activeStarterPack.uri,
+        })
+        starterPack = spRes.data.starterPack
+
+        if (starterPack.list) {
+          const listRes = await agent.app.bsky.graph.getList({
+            list: starterPack.list.uri,
+            limit: 50,
+          })
+          listItems = listRes.data.items
+        }
+      }
```

Comment:
Makes sense to do this up front, and we haven't done much error handling here before, but what do you think about more gracefully handling failures of this data separately?

Like if we get this data up front, _then_ handle it later on. But if this data fails, existing code continues to run as it does today.

Reply:
not opposed. best probably to get the user through signup successfully. i can think of a case where a deletion might happen before the user even gets through signup - and if that happens then they're screwed until they decide to not use the starter pack 😅  good call

Reply:
[`e9dcb8d` (#4548)](https://github.com/bluesky-social/social-app/pull/4548/commits/e9dcb8d5ed73a3897a7e2105bf6d688995bc1b8e)